src/components/PeopleTable.tsx

```diff
@@ -1,5 +1,113 @@
+import { Link, useLocation, useSearchParams } from 'react-router-dom';
+import { Person } from '../types';
+import { PersonLink } from './PersonLink';
+import { useEffect, useState } from 'react';
+import { useSortIcons } from '../utils/hooks/useSortIcons';
+import { getCentury } from '../utils/getCentury';
+
 /* eslint-disable jsx-a11y/control-has-associated-label */
-export const PeopleTable = () => {
+
+interface Props {
+  peopleFromServer: Person[];
+}
+
+export const PeopleTable: React.FC<Props> = ({ peopleFromServer }) => {
+  const [sortedList, setSortedList] = useState<Person[]>(peopleFromServer);
+  const [searchParams] = useSearchParams();
+  const location = useLocation();
+
+  const sortParam = searchParams.get('sort');
+  const orderParam = searchParams.get('order');
+
+  const sortClasses = useSortIcons({
+    sortParam,
+    orderParam,
+  });
+
+  function addSortParams(sortBy: string) {
+    const search = new URLSearchParams(searchParams);
+
+    if (search.get('sort') !== sortBy) {
+      search.set('sort', sortBy);
+      search.delete('order');
+    } else if (!search.has('order')) {
+      search.set('order', 'desc');
+    } else {
+      search.delete('sort');
+      search.delete('order');
+    }
+
+    return search.toString();
+  }
+
+  useEffect(() => {
+    let sorted = [...peopleFromServer];
+
+    switch (sortParam) {
+      case 'name':
+        sorted.sort((a, b) => a.name.localeCompare(b.name));
+        break;
+      case 'sex':
+        sorted.sort((a, b) => {
+          if (a.sex === 'm' && b.sex === 'f') {
+            return -1;
+          }
+
+          if (a.sex === 'f' && b.sex === 'm') {
+            return 1;
+          }
+
+          return 0;
+        });
+        break;
+      case 'born':
+        sorted.sort((a, b) => a.born - b.born);
+        break;
+      case 'died':
+        sorted.sort((a, b) => a.died - b.died);
+        break;
+      default:
+        sorted = [...peopleFromServer];
+    }
+
+    if (orderParam === 'desc') {
+      sorted.reverse();
+    }
+
+    setSortedList(sorted);
+  }, [peopleFromServer, searchParams, orderParam, sortParam]);
+
+  useEffect(() => {
+    const sexParam = searchParams.get('sex');
+    const queryParam = searchParams.get('query');
+    const centuryParams = searchParams.getAll('centuries').map(d => +d);
+
+    if (sexParam) {
+      setSortedList(curr => curr.filter(person => person.sex === sexParam));
```

Comment:
The filtering logic for sex is applied directly to the current sorted list, which may override previous sorting or filtering results. Consider applying all filters in a single `useEffect` to ensure the order of operations is consistent and predictable.